### ISSUE TITLE
axes selection in axes editor

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#5468]
Currently, when there are multiple axes, the curves line and axes editor (aka the green tick button) first pops a window to select the axes, with entries such as
```
title (<matplotlib.axes._subplots.AxesSubplot object at 0x...>) # if there is a title
<matplotlib.axes._subplots.AxesSubplot object at 0x...> # if there is no title
```

I would suggest switching this to something more user-friendly such as
```
title
<anonymous axes>
```
possibly keeping the class name and object id as tooltips.

Thoughts?

**Comment by mdboom** [comment]
Cc: @mfitzp

**Comment by tacaswell** [comment]
:+1:

**Comment by anntzer** [comment]
While working on this I realized that the selection dialog (`backend_qt5.py` lines 620+) would default to a combination of the axes title, axes ylabel, axes label and axes repr (depending on what is available).  In particular I had not noticed that axes can have a label set by `set_label` (like any artist).  Is this label used anywhere else?

**Comment by anntzer** [comment]
Closed by #5667.
